dataclients/kubernetes/clusterstate.go

```diff
@@ -83,38 +84,35 @@ func (state *clusterState) GetEndpointsByService(namespace, name, protocol strin
 	return targets
 }
 
-// GetEndpointsByName returns the skipper endpoints for kubernetes endpoints or endpointslices.
-// This function works only correctly for endpointslices (and likely endpoints) with one port with the same protocol ("TCP", "UDP").
-func (state *clusterState) GetEndpointsByName(namespace, name, protocol, scheme string) []string {
-	epID := endpointID{
-		ResourceID: newResourceID(namespace, name),
-		Protocol:   protocol,
-	}
```

Comment:
This was wrong because it does not contain target unlike other places that use state.cachedEndpoints